### PR TITLE
tobiko-app image size reduced from 3GB to 1.3GB

### PR DESCRIPTION
**Pull request by singulart**
Hey, Team! Came across your repository and thought I could help in optimizing the Docker image size.

### CODE REVIEW

File: Dockerfile.api [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
+RUN apt-get update && apt-get -y install libpq-dev gcc
 COPY setup.py setup.py
 COPY sqlmesh/_version.py sqlmesh/_version.py
+ENV VIRTUAL_ENV=/.venv
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+RUN python3 -m venv $VIRTUAL_ENV
+RUN pip3 install -e .[dev,web]
 
-RUN pip install -e .[dev,web]
+FROM python:3.11-slim-bullseye as app-image
```
Comment (tobymao) [inline Dockerfile.api:10]: why are you referencing python 3.11 both times? can you explain a bit about what this pr is doing?

Comment (singulart) [reply Dockerfile.api:10]: The Dockerfile was updated the following way:
1) The "slimmer" base image `python:3.11-slim-bullseye` was used.
2) Multi-stage build approach was used: https://docs.docker.com/build/building/multi-stage

The #2 is why the image is being referenced twice. Hope that helps! Let me know if you need more clarity on this ;)

**Comment by tobymao** [comment]
@singulart thanks for the PR. since this docker image is used for development and never shipped to the user, I don't think we need to add complexity to reduce the size of the image.

so for now i'm going to close this pr.

**Comment by singulart** [comment]
> @singulart thanks for the PR. since this docker image is used for development and never shipped to the user, I don't think we need to add complexity to reduce the size of the image.
> 
> so for now i'm going to close this pr.

Sure @tobymao! Should the Docker image size become an issue going forward, you know how to solve it. Good luck!

**Comment by tobymao** [comment]
thanks so much

**Pull request closed without merging**